### PR TITLE
feat: Test Queue v2

### PR DESCRIPTION
**Pull request by howard-e**
This PR provides an updated version of the Test Queue as described at https://github.com/w3c/aria-at-app/issues/791#issuecomment-2002610093. It is available on the sandbox for live testing at [`/test-queue`](https://aria-at-app-sandbox.bocoup.com/test-queue). The previous version is still available at [`/test-queue-old`](https://aria-at-app-sandbox.bocoup.com/test-queue-old).

In advancing this work, found that were new design expectations of some of the previously used Test Queue components, found it easier to copy over and make the tweaks rather than bring destructive changes to the old test queue's pieces (and to also verify not breaking expected functionality). As such, there is additional cleanup that will come out of this change such as removing the old test queue or the copied over components, but I'd rather not inflate this PR further and instead follow up in a subsequent PR.

### CODE REVIEW

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (stalgiag) [inline client/components/ManageTestQueue/ManageAtVersions.jsx:38]: Is this used to prevent state updates on initial render? Just noting that this struck me as odd.

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (stalgiag) [inline client/components/ManageTestQueue/ManageAtVersions.jsx:121]: Curious why this pattern is used. Shorter independent functions for each one of these `type`s could improve readability

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (howard-e) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:38]: Yep. This was originally moved from `ManageTestQueue/index.js` and there was an attempt to remove it, but found that re-renders were being significantly caused with several actions so opted to just restore that for now.

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (howard-e) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:121]: Agreed, also a copy over but no issue with addressing before landing.

**Review by stalgiag — CHANGES_REQUESTED** [review]
Looking great! This is a major UI improvement. I left some comments inline. Most are just code quality and organization questions. I also recognize that much of this code wasn't written by others so not expecting you to speak to all of these things. I just wanted to document them for possible tech debt tasks.

I noticed some keyboard interaction issues with the "Add Test Plan to Test Queue" form. Two seem unique to this PR but otherwise happy to approve this as long as we make a list of tech debt follow-ups.

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (stalgiag) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:1]: This item "Unable to open dropdowns with Enter, able to open with VO action key" is true on main so it doesn't need to be addressed with this PR. I would expect this element to interact in the same way as the APG's "Select only Combobox" but "Enter" has no effect.

**Review by howard-e — COMMENTED** [review]
On Safari, the Exact and Minimum AT Version text leaves their respective radiogroup sections

screenshot showing the exact and minimum version text going beyond the bounds of their containers in the radiogroup

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (howard-e) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:1]: > I noticed some issues with keyboard interaction on latest Chrome.
> 
> * Unable to open dropdowns with Enter, able to open with VO action key

Not opening with `Enter` doesn't seem to be new when comparing against what's on `main` so I'd like us to address this later if needed. But I also can't tell from a quick look on the [<select> spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-select-element) if this is the expectation -- but in quickly testing a simple example across Safari, Chrome and Firefox, `Enter` isn't handled in any, but `Space` is. Is there a reason we should need to handle that?

> * Unable to tab focus to unselected "Minimum Version" radio

Looks like that's the [expected behavior for radiogroups](https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio-activedescendant/#ex_label) 

> * Unable to tab focus to "Select AT Version" or "Select a Browser" dropdown

Hmmm! I suppose that decision was made to somewhat "improve" the experience since the AT being selected is what impacts which version options and browser options can actually be displayed. Do you think this way is more confusing than not?

Comment (howard-e) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:1]: > This item "Unable to open dropdowns with Enter, able to open with VO action key" is true on main so it doesn't need to be addressed with this PR. I would expect this element to interact in the same way as the APG's "Select only Combobox" but "Enter" has no effect.

Ah should have read ahead! Okay I understand your above point

File: client/hooks/useForceUpdate.js
Comment (howard-e) [reply client/hooks/useForceUpdate.js:1]: Good point. I can certainly look more into why this way

File: client/components/ManageTestQueue/ManageAtVersions.jsx
Comment (stalgiag) [reply client/components/ManageTestQueue/ManageAtVersions.jsx:1]: > Looks like that's the [expected behavior for radiogroups](https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio-activedescendant/#ex_label)

Thanks for this! I looked at that example but for some reason, my brain didn't catch the tab vs arrow interaction working across both.

> Hmmm! I suppose that decision was made to somewhat "improve" the experience since the AT being selected is what impacts which version options and browser options can actually be displayed. Do you think this way is more confusing than not?

This makes sense! I didn't fully understand the way it was supposed to be used but with that in mind it is an improvement. Thanks for clarifying.

> Ah should have read ahead! Okay I understand your above point

Haha I should've went back and edited with a strikethrough. But yes agree that this should happen elsewhere.

I've tested a bunch at this point and feel comfortable saying that any issues I haven't uncovered yet can be fixed in a follow-up. I also feel fine with leaving any or all of the code structure improvements for follow-ups. I'll leave it to you which parts need to land now vs be noted as tech debt. I'll approve. Great work!

**Review by stalgiag — APPROVED** [review]
Approving! [See this comment for more.](https://github.com/w3c/aria-at-app/pull/1124#discussion_r1645147665)

File: client/components/TestQueue2/CompletionStatusListItem/index.jsx
Comment (howard-e) [inline client/components/TestQueue2/CompletionStatusListItem/index.jsx:1]: This file is modified version of [client/components/TestQueueCompletionStatusListItem/index.js](https://github.com/w3c/aria-at-app/blob/c786a3ef5f360c77169fce32e75a0cc27a01d9f1/client/components/TestQueueCompletionStatusListItem/index.js)

File: client/components/TestQueue2/CompletionStatusListItem/BotTestCompletionStatus/index.js
Comment (howard-e) [reply client/components/TestQueue2/CompletionStatusListItem/BotTestCompletionStatus/index.js:1]: @stalgiag Thanks again for pointing this out, I've pushed https://github.com/w3c/aria-at-app/pull/1124/commits/c85e49487f02b67acd8e89cce7d1eb1edbc3489d to address this

**Comment by howard-e** [comment]
> On Safari, the Exact and Minimum AT Version text leaves their respective radiogroup sections
> 
> screenshot showing the exact and minimum version text going beyond the bounds of their containers in the radiogroup

This should be addressed by https://github.com/w3c/aria-at-app/pull/1124/commits/c04e71e19463a588f142e8e9172fce146ccc1cf1

**Comment by howard-e** [comment]
> > On Safari, the Exact and Minimum AT Version text leaves their respective radiogroup sections
> > screenshot showing the exact and minimum version text going beyond the bounds of their containers in the radiogroup
> 
> This should be addressed by [c04e71e](https://github.com/w3c/aria-at-app/commit/c04e71e19463a588f142e8e9172fce146ccc1cf1)

Reverted this due to a visual problem -- will follow up in a subsequent change to not block on this

**Pull request merged**
